latch_cli/snakemake/workflow.py

```diff
@@ -496,34 +482,41 @@ def get_param_code(
                 meta.path,
                 meta.download,
                 meta.config,
-                blob_params,
+                param_path,
             )
 
         code_blocks: List[str] = []
         if is_list_type(t):
-            code_blocks.append(dedent(f"""
-                    if len({param}) != {len(meta)}:
-                        raise ValueError("The size of input files list must be equal to the number of `SnakemakeFileMetadata` objects provided in latch_metadata")
-                    """))
+            sub_typ = get_args(t)[0]
```

Comment:
idr if you guard against generic lists above this but if you don't may wanna put that here